docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+# Working with CHIP Tool
+
+The CHIP Tool (`chip-tool`) is a tool that allows to commission a Matter device
+into the network and to communicate with it using the Zigbee Cluster Library
+(ZCL) messages.
```

Comment:
Although the current text actually does not lie (we are just not mentioning that apart from what is written, the tool has also other minor capabilities :) ), I will try to figure out more generic description.

But it's true that describing general application flow is rather no-go because it strictly depends on the command/sub-command/option combination.